pkg/gardener/chartrenderer/chartrenderer.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chartrenderer
+
+import (
+	"github.com/gardener/gardener/pkg/chartrenderer"
+	"k8s.io/client-go/rest"
+)
+
+type factory struct{}
```

Comment:
Instead of an empty struct, I would create a function type satisfying the Factory interface:

```go
type FactoryFunc func(*rest.Config) (chartrenderer.Interface, error)

func (f FactoryFunc) NewForConfig(config *rest.Config) (chartrenderer.Interface, error) {
	return f(config)
}
```

And then use the `chartrenderer.NewForConfig` function as the implementation.

Reply:
Same question as for the client factory function: Why expose an additional type that is not that easily extendable? If we never show how the factory is implemented, why is it important to have this function type?

Reply:
See above. Although it's relatively minor, IMO an adapter type captures the intention to implement the interface via a specific function (or a set of function) better by making it more explicit. It can always be refactored if it should be extended at some point.